components/scream/src/share/util/scream_common_physics_functions_impl.hpp

```diff
@@ -208,6 +208,59 @@ void PhysicsFunctions<DeviceT>::calculate_z_int(const MemberType& team,
   column_ops::template column_scan<FromTop>(team,num_levs,dz,z_int,z_surf);
 }
 
+template<typename DeviceT>
+template<typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+ScalarT PhysicsFunctions<DeviceT>::calculate_vmr_from_mmr(const Real& gas_mol_weight, const ScalarT& mmr)
+{
+  using C = scream::physics::Constants<Real>;
+  constexpr Real air_mol_weight   = C::MWdry;
+
+  return mmr / (1.0 - mmr) * air_mol_weight/gas_mol_weight;
+
+}
+
+template<typename DeviceT>
+template<typename ScalarT, typename InputProviderX>
+KOKKOS_INLINE_FUNCTION
+void PhysicsFunctions<DeviceT>::calculate_vmr_from_mmr(const MemberType& team,
+                                                       const Real gas_mol_weight,
+                                                       const InputProviderX& mmr,
+                                                       const view_1d<ScalarT>& vmr)
+{
+  Kokkos::parallel_for(Kokkos::TeamThreadRange(team,vmr.extent(0)),
+                       [&] (const int k) {
+    vmr(k) = calculate_vmr_from_mmr(gas_mol_weight,mmr(k));
+  });
+}
+
+template<typename DeviceT>
+template<typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+ScalarT PhysicsFunctions<DeviceT>::calculate_mmr_from_vmr(const Real& gas_mol_weight, const ScalarT& vmr)
+{
+  using C = scream::physics::Constants<Real>;
+  constexpr Real air_mol_weight   = C::MWdry;
+  const Real mol_weight_ratio = gas_mol_weight/air_mol_weight;
+
+  return mol_weight_ratio*vmr / (1.0 + mol_weight_ratio*vmr); 
```

Comment:
This does not look quite right either. With the fix above, I think we want:
```
mmr = vmr * (1 - qv) * gas_mol_weight / air_mol_weight;
```
where `qv` is the wet mixing ratio of water vapor.